modules/ecs-cluster-ec2-asg/v0.0.1/variables.tf

```diff
@@ -0,0 +1,281 @@
+# ASG related variables
+# Required:
+variable "security_group_ids" {
+  description = "List of security group IDs to assign to instances"
+  type        = list(string)
+}
+
+variable "subnet_ids" {
+  description = "List of VPC Subnet IDs to launch instances in"
+  type        = list(string)
+}
+
+# Optional:
+variable "instance_type" {
+  description = "See: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#AvailableInstanceTypes"
+  type        = string
+  default     = "t2.micro"
+}
+
+variable "user_data" {
+  description = "Bash code for inclusion as user_data on instances. By default contains minimum for registering with ECS cluster"
+  type        = string
+  default     = "false"
+}
+
+variable "min_size" {
+  description = "Minimum size of the Auto Scaling Group"
+  type        = number
+  default     = 1
+}
+
+variable "max_size" {
+  description = "Maximum size of the Auto Scaling Group"
+  type        = number
+  default     = 5
+}
+
+variable "desired_capacity" {
+  description = "Number of Amazon EC2 instances that should be running in the group"
+  type        = number
+  default     = 1
+}
+
+variable "health_check_type" {
+  description = "Controls how health checking is done. Valid values are EC2 and ELB"
+  type        = string
+  default     = "EC2"
+  validation {
+    condition     = contains(["EC2", "ELB"], var.health_check_type)
```

Comment:
Does ELB make sense in the context of ECS? because the instances of the ASG themselves will never be load balanced. I think for this usecase it should always be EC2